### ISSUE TITLE
`state_getKeys` wrongly returns an empty list

### ISSUE THREAD

**Issue by josepot** [smol-dot/smoldot#174]
I've created [this sandbox on stackblitz](https://stackblitz.com/edit/node-9wgtdu?file=index.js) to reproduce and isolate the issue.

In short, the issue is that smoldot -while connected to the polkadot network- returns an empty list of keys as a response to the following RPC request (which, in this example, asks for the keys of the `staking.validators` storage entry):

```json
{
  "id":1,
  "jsonrpc":"2.0",
  "method":"state_getKeys",
  "params":["0x5f3e4907f716ac89b6347d15ececedca88dcde934c658227ee1dfafcd6e16903"]
}
```

while the exact same RPC request against `wss://apps-rpc.polkadot.io/` returns a very long list of keys.

**Comment by rossbulat** [comment]
Great, I experienced the same issue in the staking dashboard when fetching `nominationPools.bondedPools`, `nominationPools.poolMembers` also with Polkadot JS API.

**Comment by tomaka** [comment]
Received proofs: [logs.txt](https://github.com/smol-dot/smoldot/files/10721214/logs.txt)

**Comment by tomaka** [comment]
I just wanted to point out that `state_getKeys` takes a long time because we're waiting for https://github.com/paritytech/substrate/issues/10623

**Comment by josepot** [comment]
> I just wanted to point out that `state_getKeys` takes a long time to execute, and that's normal. The fix is [paritytech/substrate#10623](https://github.com/paritytech/substrate/issues/10623), but it's taking ages to land.
> 
> After [paritytech/substrate#10623](https://github.com/paritytech/substrate/issues/10623), #177 will improve the execution time further.

Thanks for looking into this @tomaka and also for sharing all this context. Much appreciated!

I always assumed that it would take a rather long time for a light-client to resolve a `state_getKeys` request. That makes sense, ofc. However, what's currently happening is that smoldot resolves the request with an empty list.

So, are you saying that it's not possible for smoldot to resolve the `state_getKeys` requests (at all) until  https://github.com/paritytech/substrate/issues/10623 is solved? Or that it's possible to solve this issue before at the expense of having to wait for a very long time to get the response?
